sherpa/tests/test_data.py

```diff
@@ -141,22 +148,23 @@ def test_base_data_instantiation():
         BaseData()
 
 
-@pytest.mark.xfail(reason="DataND did not serve any purpose and had a misleading name")
-def test_base_datand_instantiation():
-    DataND()
+@pytest.mark.parametrize("data", (Data, Data2D, Data2DInt), indirect=True)
+def test_data_get_x(data):
+    with pytest.raises(AttributeError):
+        data.get_x()
 
 
-@pytest.mark.xfail(reason="methods did not belong and were removed")
-@pytest.mark.parametrize("data", (Data, Data2D, Data2DInt, Data1DInt), indirect=True)
-def test_data_get_x(data):
-    with pytest.raises(NameError):
+@pytest.mark.xfail
+@pytest.mark.parametrize("data", (Data1DInt, ), indirect=True)
+def test_data_get_x_special(data):
+    # XFAIL: These classes still provide get_x
+    with pytest.raises(AttributeError):
         data.get_x()
```

Comment:
Note (not for this PR): I do see an argument for providing a `get_x` for all 1d classes. For the binned classes (Data1DInt) it would provide a [2,n] array or a tuple or something, but it is confusing that there are so many way to access the x data with partially overlapping names and functionality (#1174)

Reply:
So, this was removed in #614 and I don't think we want to go back on that work without a lot of careful thinking.